enclave/CMakeLists.txt

```diff
@@ -1,6 +1,8 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
+add_subdirectory(crypto)
```

Comment:
@andschwa `enclave/core` is currently getting built by root CMakeLists.txt but now `enclave` has a dependency on `enclave/crypto` as well and has to build it as part of its CMakeList.txt? This is getting increasingly awkward; should this be refactored out into a subfolder instead of keeping it at the root of /enclave? (e.g. even `enclave/enclave` works)

Reply:
🤷‍♂

Reply:
I thought oecore was built by `enclave/core/CMakeLists.txt`...

Reply:
@andschwa I'm referring to the fact that for apparent ordering reasons we end up doing `add_subdirectory(enclave/core)` in /CMakeLists.txt rather than /enclave/CMakeLists.txt (which is weird because it also has to build oeenclave.

Reply:
Oh that is weird. I don't remember doing that.

Reply:
Honestly I'd just try changing that; it's probably not necessary.

Reply:
since `3rdparty` depends on `oecore`, doesn't the `oecore` target need to be available before CMake goes in the third party directories?

Reply:
😬 Only sometimes...CMake has differing rules for when things need to be declared for its different entities. I'd have to test it to see.